### PR TITLE
initial setup the Project

### PR DESCRIPTION
**Pull request by melkamu12**
### Bookstore: initialize project with components

In this PR, I added the initial setup for bookstore.

 ## Activities done
 *  [ ] Setup a new React application using [Create React App](https://reactjs.org/docs/create-a-new-react-app.html#create-react-app)
 * [ ] Create components inside of the /components directory:
    - [ ]  Contains individual book state, as well as a button to delete the book
    - [ ]  Contains a list to renders individual books
    - [ ]  Contains a form to create new books
    - [ ] Contains a navigation to link to the 2 routes you created
* [ ] I added 2 routes in my application:
        -  /
        -  /categories

 
 # Checklist:
 * [ ] Follows the style guidelines of this project

 * [ ] There is no linters error in my code
 * [ ] follow gitflow

### CODE REVIEW

File: src/components/bookList.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const bookList = () => <div>List of the book</div>;
+export default bookList;
```
Comment (brainconnect93) [inline src/components/bookList.js]: - [ ] Great effort working on the Bookstore, Initialize project with component milestone, but according to the requirement there are some thing you are currently missing. The requirement state that

1. A `<Books>` component, which contains the following components:
   - `<BookList>` which Renders a list of books using the `<Book>` component

2. `<Book>` which Takes the book data as a prop and renders the title and the author
   - Has a button for removing books: the button wouldn't remove the books yet, this is just a placeholder for now

So therefore kindly create some list of book using the <Book> component and also let it have a remove button but not necessary it is functioning at the moment.

File: src/components/addForm.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+const addBook = () => <div>Add book form</div>;
+export default addBook;
```
Comment (brainconnect93) [inline src/components/addForm.js]: - [ ] According to the requirement, I would kindly suggest you work on this component as well so as to go inline with what we have in the project requirement

1. `<CreateBookForm>`
Should: 
* Renders a form with inputs for the title and the author of the book
* Has a button to submit
  * the button wouldn't add the book *yet*, this is just a placeholder for now

please kindly implement this in your project so as to have the proper output in your project.

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @melkamu12, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## To Highlight👍

- Descriptive Pull Request title and summary. ✔️
- `react-router-dom` should be present in `package.json` ✔️
- `React Router` has been correctly set up in `<App>` ✔️
- Good use of Git flow ✔️

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/bookList.js [hunk 3]
```diff
@@ -1,2 +1,15 @@
...
+
+const BookList = ({ Title, author }) => (
+  <div>
+    <li>
+      <h2>{Title}</h2>
+      <p>{author}</p>
+    </li>
+  </div>
+);
+BookList.propTypes = {
```
Comment (binhussen) [inline src/components/bookList.js:14]: - Great job for implementing previous reviewer suggestions, I think the `booklist` component has to render using the `book` component, so create the book component the accept book object, and render it and your `booklist` component will a list of books and pass to `book` component 👍 
```
 {books.map((book) => (
          <Book key={book.d} book={book} />
        ))}
```

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @melkamu12 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Highlights

- Well Structured Code ✔️
- Descriptive pull request ✔️
- Used Git flow ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @melkamu12 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
React component used to render a list of books using the Book component 👏 
Passed each `book's `details as props ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**